python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_restraint.py

```diff
@@ -380,7 +380,7 @@ def system(self, system):
         # Store a copy of solvated system.
         self._system = system.copy()
 
-    def _gromacs_boresch(self, perturbation_type=None):
+    def _gromacs_boresch(self, perturbation_type=None, restraint_lambda=False):
```

Comment:
This function is getting too complex and it probably warrants its own module/class tbh. Not a huge fan of a large number of nested functions.

Reply:
I cannot say that I like this massive function. But the other three restraints are using the same function type interface, so it would be for the other three, you call a function and for this one, you generate a object and get string from it.